### PR TITLE
LocationFetcher updated to support API 23 (M).

### PR DESCRIPTION
**Pull request by teamagam1**
LocationFetcher updated to support API 23 (M).

The registration to the location fetcher was moved to the Launcher Activity.

### CODE REVIEW

**Comment by review-ninja** [comment]
<a href="https://app.review.ninja/teamagam/GimelGimel/pull/15" target="_blank">ReviewNinja

File: app/src/main/java/com/teamagam/gimelgimel/app/GGApplication.java [hunk 1]
```diff
@@ -39,41 +30,17 @@ public void onCreate() {
 
         CheckIfAppUpdated();
```
Comment (gRaytan) [inline app/src/main/java/com/teamagam/gimelgimel/app/GGApplication.java:31]: mPrefSecureKey
what is it?
should be deleted

Comment (gRaytan) [inline app/src/main/java/com/teamagam/gimelgimel/app/GGApplication.java:31]: CheckIfAppUpdated - all in comments, why?

File: app/src/main/java/com/teamagam/gimelgimel/app/GGApplication.java [hunk 2]
```diff
@@ -39,41 +30,17 @@ public void onCreate() {
...
-
-            @Override
-            public void onNewLocationSample(LocationSample locationSample) {
-                GGMessagingUtils.sendUserLocationMessageAsync(locationSample);
-                Message msg = new MessageUserLocation(NetworkUtil.getMac(), locationSample);
-                MessageBroadcastReceiver.sendBroadcastMessage(GGApplication.this, msg);
-            }
-        });
-        mLocationFetcher.addProvider(LocationFetcher.ProviderType.LOCATION_PROVIDER_GPS);
-//        mLocationFetcher.addProvider(LocationFetcher.ProviderType.LOCATION_PROVIDER_NETWORK);
```
Comment (gRaytan) [inline app/src/main/java/com/teamagam/gimelgimel/app/GGApplication.java:67]: in comment, we need it or not?

File: app/src/main/java/com/teamagam/gimelgimel/app/GGApplication.java [hunk 3]
```diff
@@ -39,41 +30,17 @@ public void onCreate() {
...
-//        mLocationFetcher.addProvider(LocationFetcher.ProviderType.LOCATION_PROVIDER_PASSIVE);
-        mLocationFetcher.registerForUpdates(locationMinUpdatesMs, locationMinDistanceM);
+
 
     }
 
     @Override
     public void onTerminate() {
         super.onTerminate();
-        mLocationFetcher.unregisterFromUpdates();
```
Comment (gRaytan) [inline app/src/main/java/com/teamagam/gimelgimel/app/GGApplication.java:76]: you sure tahts the right order?

Comment (gRaytan) [inline app/src/main/java/com/teamagam/gimelgimel/app/GGApplication.java:76]: never mind

File: app/src/main/java/com/teamagam/gimelgimel/app/GGApplication.java (hunk 2 above)
Comment (gRaytan) [inline app/src/main/java/com/teamagam/gimelgimel/app/GGApplication.java:67]: never mind

File: app/src/main/java/com/teamagam/gimelgimel/app/control/sensors/LocationFetcher.java [hunk 4]
```diff
@@ -23,8 +27,6 @@
  */
 public class LocationFetcher {
 
-    private static final String LOG_TAG = LocationFetcher.class.getSimpleName();
```
Comment (gRaytan) [inline app/src/main/java/com/teamagam/gimelgimel/app/control/sensors/LocationFetcher.java:26]: y? not in use? how will we write logs?

**Pull request merged**